File: app/src/main/java/hibernate/v2/testyourandroid/ui/adapter/MonitorFragmentPagerAdapter.java
package hibernate.v2.testyourandroid.ui.adapter;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.view.View;
import android.widget.TextView;

import hibernate.v2.testyourandroid.R;
import hibernate.v2.testyourandroid.ui.fragment.MonitorCpuFragment;
import hibernate.v2.testyourandroid.ui.fragment.MonitorMemoryFragment;
import hibernate.v2.testyourandroid.ui.fragment.MonitorNetworkFragment;

public class MonitorFragmentPagerAdapter extends FragmentPagerAdapter {

	private String tabTitles[];

	private Context mContext;

	/**
	 * Constructor of the class
	 */
	public MonitorFragmentPagerAdapter(FragmentManager fm, Context mContext) {
		super(fm);
		tabTitles = mContext.getResources().getStringArray(R.array.test_monitor_tab_title);
		this.mContext = mContext;
	}

	/**
	 * Returns the number of pages
	 */
	@Override
	public int getCount() {
		return 3;
	}

	/**
	 * This method will be invoked when a page is requested to create
	 */
	@Override
	public Fragment getItem(int position) {
		Fragment fragment;
		switch (position) {
			/* Android tab is selected */
			case 0:
				fragment = new MonitorCpuFragment();
				break;
			case 1:
				fragment = new MonitorNetworkFragment();
				break;
			case 2:
				fragment = new MonitorMemoryFragment();
				break;
			default:
				fragment = null;
		}
		return fragment;
	}

	public View getTabView(int position) {
		View v = View.inflate(mContext, R.layout.custom_tab, null);
		TextView tv = v.findViewById(R.id.tabTitleTv);
		tv.setText(tabTitles[position]);
		return v;
	}
}

File: app/src/main/java/hibernate/v2/testyourandroid/ui/info/monitor/MonitorFragmentPagerAdapter.kt
package hibernate.v2.testyourandroid.ui.info.monitor

import android.view.LayoutInflater
import android.view.View
import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import hibernate.v2.testyourandroid.R
import hibernate.v2.testyourandroid.databinding.CustomTabBinding

class MonitorFragmentPagerAdapter(
    private val fragment: Fragment
) : FragmentStateAdapter(fragment) {

    private val tabTitles: Array<String> =
        fragment.resources.getStringArray(R.array.test_monitor_tab_title)

    override fun getItemCount(): Int = 3

    override fun createFragment(position: Int): Fragment {
        return when (position) {
            1 -> MonitorNetworkFragment()
            2 -> MonitorMemoryFragment()
            0 -> MonitorCPUFragment()
            else -> MonitorCPUFragment()
        }
    }

    fun getTabView(position: Int): View {
        val viewBinding = CustomTabBinding.inflate(LayoutInflater.from(fragment.context))
        viewBinding.tabTitleTv.text = tabTitles[position]
        return viewBinding.root
    }

}